### ISSUE TITLE
Add support for tag completion

### ISSUE THREAD

**Issue by realprogrammersusevim** [artempyanykh/marksman#111]
I often use tags to help me find notes in my Zettelkasten system and it would be extremely useful to have some sort of tag completion. It could be triggered by typing a `#` and then a letter to show that the user isn't trying to type a header. Then the completion options would be all the other tags used in a folder. This would help to show what other common tags are and make sure all of them are uniformly spelled and worded.

**Comment by artempyanykh** [comment]
Thanks for raising this @realprogrammersusevim! Could you provide a small snippet of markdown doc showing how you use tags?

**Comment by realprogrammersusevim** [comment]
Here's an example of how I put tags in my documents:

```markdown
# My Note

Saying lot's of smart stuff. Much smartness, much wow.

[[Another Note]]

#tag #smartness #anotherTag #brilliantInsight
```

**Comment by BonaBeavis** [comment]
Another way to define tags is using markdown frontmatter:
```markdown
---
tags:
  - foo
  - bar
---

**Comment by pidgeon777** [comment]
Hello, any update on this useful feature?

**Comment by artempyanykh** [comment]
Hi @pidgeon777! Well, winter holidays wasn't the most productive season :) Anyhow, I plan to pick this up soon, although the first version won't support tags defined in front-matter, only `#tagname` will be supported initially.

**Comment by artempyanykh** [comment]
Support for tags has been merged in #159 . I'll push the next release soon-ish, but feel free to build from source and give it a spin meantime.

**Comment by eguiraud** [comment]
Hi, I think this is available in the latest release and can be closed.
